src/vs/workbench/contrib/positronNotebook/browser/positronNotebook.contribution.ts

```diff
@@ -75,52 +78,48 @@ class PositronNotebookContribution extends Disposable {
 			},
 			{
 				createEditorInput: async ({ resource, options }) => {
-					// Determine notebook type from file content or metadata
-					const viewType = await this.detectNotebookViewType(resource);
-
-					// Type guard for backup working copy options
-					interface BackupWorkingCopyOptions {
-						_backupId?: string;
-						_workingCopy?: IWorkingCopyIdentifier;
-					}
-					function hasBackupWorkingCopyOptions(obj: unknown): obj is BackupWorkingCopyOptions {
-						return typeof obj === 'object' && obj !== null &&
-							('_backupId' in obj || '_workingCopy' in obj);
-					}
-
-					// Preserve backup options if they exist
-					const positronOptions: PositronNotebookEditorInputOptions = {
-						startDirty: false,
-						_backupId: hasBackupWorkingCopyOptions(options) ? options._backupId : undefined,
-						_workingCopy: hasBackupWorkingCopyOptions(options) ? options._workingCopy : undefined
-					};
-
-					const editorInput = PositronNotebookEditorInput.getOrCreate(
+					const notebookEditorInput = PositronNotebookEditorInput.getOrCreate(
 						this.instantiationService,
 						resource,
 						undefined,
-						viewType,
-						positronOptions
 					);
-
-					return { editor: editorInput, options };
+					return { editor: notebookEditorInput, options };
 				}
-			}
+			},
 		));
-	}
-
-	private async detectNotebookViewType(resource: URI): Promise<string> {
-		// Check if there's already an open notebook model for this URI
-		const existingModel = this.notebookService.getNotebookTextModel(resource);
-		if (existingModel) {
-			return existingModel.viewType;
-		}
-
-		// Use NotebookService to detect the correct viewType
-		const notebookProviders = this.notebookService.getContributedNotebookTypes(resource);
 
-		// Default to jupyter-notebook if detection fails
-		return notebookProviders[0]?.id || 'jupyter-notebook';
+		// Register for cells in .ipynb files
+		this._register(this.editorResolverService.registerEditor(
+			`${Schemas.vscodeNotebookCell}:/**/*.ipynb`,
+			notebookEditorInfo,
+			{
+				singlePerResource: true,
+				canSupportResource: (resource: URI) => {
+					return resource.scheme === Schemas.vscodeNotebookCell;
+				}
+			},
+			{
+				createEditorInput: async (editorInput) => {
+					const parsed = CellUri.parse(editorInput.resource);
```

Comment:
This is our entrypoint when a cell URI is opened, if our editor is configured as the default.